extensions/civetweb/processors/ListenHTTP.h

```diff
@@ -165,28 +192,66 @@ class ListenHTTP : public core::Processor {
      * Sets a static response body string to be used for a given URI, with a number of seconds it will be kept in memory.
      * @param response
      */
-    void setResponseBody(const ResponseBody& response);
+    bool setResponseBody(const ResponseBody& response);
+
+    bool dequeueRequest(Request& req);
+
+    size_t requestCount() const {
+      return request_buffer_.size();
+    }
+
+    bool empty() const {
+      return request_buffer_.empty();
+    }
 
-    bool dequeueRequest(FlowFileBufferPair &flow_file_buffer_pair);
+    void stop() {
+      request_buffer_.stop();
+      Request req;
+      while (dequeueRequest(req)) {
+        std::promise<void> req_done_promise;
+        auto req_done = req_done_promise.get_future();
+        req.set_value(nonstd::make_unexpected(FailureValue{Handler::FailureReason::PROCESSOR_SHUTDOWN, std::move(req_done_promise)}));
+        req_done.wait();
+      }
+    }
 
    private:
     static void sendHttp500(struct mg_connection *conn);
     static void sendHttp503(struct mg_connection *conn);
     bool authRequest(mg_connection *conn, const mg_request_info *req_info) const;
     void setHeaderAttributes(const mg_request_info *req_info, core::FlowFile& flow_file) const;
-    void writeBody(mg_connection *conn, const mg_request_info *req_info, bool include_payload = true);
-    static std::unique_ptr<io::BufferStream> createContentBuffer(struct mg_connection *conn, const struct mg_request_info *req_info);
-    void enqueueRequest(mg_connection *conn, const mg_request_info *req_info, std::unique_ptr<io::BufferStream>);
+    void writeBody(core::ProcessSession* payload_reader, mg_connection *conn, const mg_request_info *req_info);
+    void enqueueRequest(mg_connection *conn, const mg_request_info *req_info, bool write_body);
+
+    class RequestBuffer : public utils::ConcurrentQueue<Request> {
```

Comment:
I think we could avoid the inheritance and use a wrapper around the concurrent queue instead. This way we don't need to modify the ConcurrentQueue and make it's enqueue method virtual, just have the wrapper as a proxy for the few methods that are used by the Handler class. We can also replace the `running_` with and atomic bool variable so no additional locks are needed in the wrapper.

Reply:
removed the inheritance, unfortunately we have to employ a lock, as it can happen that we check `running_` on new incoming connection, which is true, then we call `notifyStop`, empty the queue, and only then enqueue the request, leaving it hanging, the ConditionalConcurrentQueue variant sadly does not solve this problem

Reply:
Thanks!